dali/dfuplus/dfuplus.cpp

```diff
@@ -721,14 +746,15 @@ int CDfuPlusHelper::despray()
 
     const char* dstxml = globals->queryProp("dstxml");
     const char* dstip = globals->queryProp("dstip");
+    const char* dstplane = globals->queryProp("dstplane");
     const char* dstfile = globals->queryProp("dstfile");
 
     bool nowait = globals->getPropBool("nowait", false);
 
     MemoryBuffer xmlbuf;
     if(dstxml == nullptr)
     {
-        if(dstip == nullptr) {
+        if (!isEmptyString(dstplane) && isEmptyString(dstip)) {
```

Comment:
Shouldn't this be isEmptyString(dstplane)?

Reply:
no.  This section should not be executed if dstplane has a value. This code section is for handling the case where the dstip is empty so assume that the destination ip is localhost.   If the dstplane has a value then the destination isn't localhost so this section should not be executed.

Reply:
Agreed - but your test *will* execute it if dstplane has a value.

Reply:
sorry, you're right.